hibernate-core/src/main/resources/org/hibernate/hibernate-mapping-4.0.xsd

```diff
@@ -395,7 +395,7 @@ arbitrary number of queries, and import declarations of arbitrary classes.
     <xs:attribute name="index" type="xs:string"/> <!-- include the columns spanned by this association in an index -->
     <xs:attribute name="insert" default="true" type="xs:boolean"/>
     <xs:attribute name="lazy" default="false" type="xs:boolean"/>
-    <xs:attribute name="meta-type" type="xs:string"/> <!--- default: Hibernate.STRING -->
+    <xs:attribute name="meta-type" default="string" type="xs:string"/> <!--- default: Hibernate.STRING -->
```

Comment:
I'm not sure the change above is actually needed. Please give it a try without your fix to find out. If it's not necessary, then please add a commit with your test and back out of this change and the change to hibernate-mapping-3.0.dtd.

Reply:
Yes, my change to 4.0.xsd is necessary as well. I'll add unit tests to it.

Reply:
Sorry, I made a stupid mistake. My change to 4.0.xsd is wrong. I'll figure it out.

Reply:
I found that modifying `hibernate-mapping-4.0.xsd` has no effect on the behavior. Instead, modifying `legacy-mapping-4.0.xsd` will actually fix this issue.